File: klinara-ios/klinara-ios/Services/Staff/StaffService.swift
import Foundation

/// Personel profili ve hizmet yetkinliği uçları (Batch 2.2).
protocol StaffService: Sendable {

    /// `GET /staff`
    func profiles() async throws -> [StaffProfile]

    /// `GET /staff/:id`
    func profile(id: String) async throws -> StaffProfile

    /// `POST /staff`
    func createProfile(_ input: CreateStaffProfileInput) async throws -> StaffProfile

    /// `PATCH /staff/:id`
    func updateProfile(id: String, _ input: UpdateStaffProfileInput) async throws -> StaffProfile

    /// `PUT /staff/:id/services` — yetkinlik listesinin tamamını değiştirir.
    func replaceSkills(id: String, _ input: ReplaceStaffServicesInput) async throws -> StaffProfile
}

struct LiveStaffService: StaffService {

    private let client: APIClient

    init(client: APIClient) {
        self.client = client
    }

    func profiles() async throws -> [StaffProfile] {
        let response: ListEnvelope<StaffProfile> = try await client.send(APIRequest.get("staff"))
        return response.data
    }

    func profile(id: String) async throws -> StaffProfile {
        try await client.send(APIRequest.get("staff/\(id)"))
    }

    func createProfile(_ input: CreateStaffProfileInput) async throws -> StaffProfile {
        try await client.send(APIRequest.post("staff", body: input))
    }

    func updateProfile(id: String, _ input: UpdateStaffProfileInput) async throws -> StaffProfile {
        try await client.send(APIRequest.patch("staff/\(id)", body: input))
    }

    func replaceSkills(id: String, _ input: ReplaceStaffServicesInput) async throws -> StaffProfile {
        try await client.send(APIRequest.put("staff/\(id)/services", body: input))
    }
}

File: klinara-android/app/src/main/java/com/klinara/android/services/staff/StaffService.kt
package com.klinara.android.services.staff

/**
 * Personel uçları — `staff:read` okur, `staff:write` yazar.
 *
 * Takvim personel adı ve rengi olmadan çizilemediği için servisin **okuma yarısı** A3.1'e
 * alınmıştı; yazma yarısı A7.2'de geldi. Silme yok: pasife almak `update(isActive = false)`.
 */
interface StaffService {
    /**
     * `GET staff` — [branchId] verilmezse kiracının tüm personeli. Verilirse (A7.4–A7.5) yalnız o
     * şubeye ait olanlar: ana şube VEYA şube üyeliği ([StaffProfile.worksIn]). Erişilemeyen
     * şube 403 `BRANCH_FORBIDDEN`.
     */
    suspend fun list(branchId: String? = null): List<StaffProfile>

    /** `GET staff/:id`. */
    suspend fun profile(id: String): StaffProfile

    /** `POST staff`. Aynı kullanıcıya ikinci profil 409. */
    suspend fun create(input: CreateStaffProfileInput): StaffProfile

    /** `PATCH staff/:id`. */
    suspend fun update(
        id: String,
        input: UpdateStaffProfileInput,
    ): StaffProfile

    /** `PUT staff/:id/services` — yetkinlik listesini TAMAMEN değiştirir. */
    suspend fun replaceSkills(
        id: String,
        input: ReplaceStaffServicesInput,
    ): StaffProfile
}
